src/citrine/informatics/predictors/graph_predictor.py

```diff
@@ -31,40 +41,75 @@ class GraphPredictor(VersionedEngineResource['GraphPredictor'], Predictor):
+    _api_version = "v3"
 
     def __init__(self,
                  name: str,
                  *,
                  description: str,
-                 predictors: List[Union[UUID, Predictor]],
+                 predictors: List[PredictorNode],
                  training_data: Optional[List[DataSource]] = None):
         self.name: str = name
         self.description: str = description
-        self.predictors: List[Union[UUID, Predictor]] = predictors
+        self.predictors: List[PredictorNode] = predictors
         self.training_data: List[DataSource] = training_data or []
 
-    def _post_dump(self, data: dict) -> dict:
-        data = super()._post_dump(data)
-        for i, predictor in enumerate(data['instance']['predictors']):
-            if isinstance(predictor, dict):
-                # embedded predictors are not modules, so only serialize their config
-                data['instance']['predictors'][i] = predictor['instance']
-        return data
-
-    @classmethod
-    def _pre_build(cls, data: dict) -> dict:
-        for i, predictor_data in enumerate(data['data']['instance']['predictors']):
-            if isinstance(predictor_data, dict):
-                data['data']['instance']['predictors'][i] = Predictor.wrap_instance(predictor_data)
-        return data
-
     def __str__(self):
         return '<GraphPredictor {!r}>'.format(self.name)
+
+    @property
+    def report(self):
+        """Fetch the predictor report."""
+        if self.uid is None or self._session is None or self._project_id is None \
+                or getattr(self, "version", None) is None:
+            msg = "Cannot get the report for a predictor that wasn't read from the platform"
+            raise ValueError(msg)
+        report_resource = ReportResource(self._project_id, self._session)
+        return report_resource.get(predictor_id=self.uid, predictor_version=self.version)
+
+    @staticmethod
+    def wrap_instance(predictor_data: dict) -> dict:
+        """Insert a serialized embedded predictor into a module envelope.
+
+        This facilitates deserialization.
+        """
+        return {
+            "data": {
+                "name": predictor_data.get("name", ""),
+                "description": predictor_data.get("description", ""),
+                "instance": predictor_data
+            }
+        }
+
+    def _path(self):
+        return format_escaped_url(
+            '/projects/{project_id}/predictors/{predictor_id}/versions/{version}',
+            project_id=self._project_id,
+            predictor_id=str(self.uid),
+            version=self.version
+        )
+
+    def predict(self, predict_request: SinglePredictRequest) -> SinglePrediction:
+        """Make a one-off prediction with this predictor."""
+        path = self._path() + '/predict'
+        res = self._session.post_resource(path, predict_request.dump(), version=self._api_version)
+        return SinglePrediction.build(res)
```

Comment:
Is this new or was it moved from somewhere else?

Reply:
This is all copied from the `Predictor` super class, which has been absorbed into the `GraphPredictor` class.